drivers/modem/hl7800.c

```diff
@@ -0,0 +1,4841 @@
+/*
+ * Copyright (c) 2020 Laird Connectivity
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT swi_hl7800
+
+#define LOG_DOMAIN modem_hl7800
+#define LOG_LEVEL CONFIG_MODEM_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_DOMAIN);
```

Comment:
BTW, this can be simplified by just writing
```
LOG_MODULE_REGISTER(modem_hl7800, CONFIG_MODEM_LOG_LEVEL);
```
so normally you do not need to define LOG_DOMAIN or LOG_LEVEL.